src/mbgl/map/transform.cpp

```diff
@@ -313,6 +313,107 @@ void Transform::_easeTo(const CameraOptions& options, double new_scale, double n
     }
 }
 
+void Transform::flyTo(const CameraOptions &options) {
```

Comment:
Can you please add documentation for all of the math things that go on here? It'd be cool to have comments that describe what you're doing on a high level.